pynbody/analysis/halo.py

```diff
@@ -481,42 +484,62 @@ def Ellipsoid(r, a,b,c, E):
         x,y,z = np.dot(np.transpose(E),[r[:,0],r[:,1],r[:,2]])
         return (x/a)**2 + (y/b)**2 + (z/c)**2
 
+
     # Define moment of inertia tensor:
-    MoI = lambda r,m: np.array([[np.sum(m*r[:,i]*r[:,j]) for j in range(3)]\
-                               for i in range(3)])
+    def MoI(r,m):
+        ShapeTensor = np.zeros((3,3))
+        for i in range(3):
+            for j in range(3):
+                ShapeTensor[i,j] = np.sum(m*r[:,i]*r[:,j])
+        return ShapeTensor
 
     # Splits 'r' array into N groups containing equal numbers of particles.
     # An array is returned with the radial bins that contain these groups.
-    sn = lambda r,N: np.append([r[i*int(len(r)/N):(1+i)*int(len(r)/N)][0]\
-                               for i in range(N)],r[-1])
+    def sn(r, N):
+        split = np.array(())
+        for i in range(N):
+            split = np.append(split, r[i*int(len(r)/N):(1+i)*int(len(r)/N)][0])
+        split = np.append(split, r[-1])
+        return split
 
     # Retrieves alignment angle:
-    almnt = lambda E: np.arccos(np.dot(np.dot(E,[1.,0.,0.]),[1.,0.,0.]))
+    def almnt(E):
+        theta = np.arccos(np.dot(np.dot(E,[1.,0.,0.]),[1.,0.,0.]))
+        return theta
     #-----------------------------FUNCTIONS-----------------------------
 
-    if (rout is None): rout = sim.dm['r'].max()
-    if (rin is None): rin = rout/1E3
+    #select particle type
```

Comment:
Using the standard pynbody family mechanism (see comment about call signature above), this would be something along the lines of:

`sim_m = sim[pynbody.family.get_family(use_family)]`

This is then completely general if people are using different family names

Reply:
Thank you for the feedback!

I have committed changes to use your suggestions.